src/authModel.js

```diff
@@ -0,0 +1,89 @@
+const Token = require('./models/auth/Token')
+const Client = require('./models/auth/Client')
+const AuthorizationCode = require('./models/auth/AuthorizationCode')
+
+module.exports = {
+  getAccessToken: accessToken => {
+    console.log('finding token, accessToken = ' + accessToken)
+    return Token.findOne({ accessToken })
+      .populate('user')
+  },
+
+  getAuthorizationCode: authCode => {
+    console.log('finding auth code, code = ' + authCode)
+    return AuthorizationCode.findOne({ authorizationCode: authCode })
+      .populate('user client')
+  },
+
+  getClient: (clientId, clientSecret) => {
+    console.log('finding client, id = ' + clientId + ', secret = ' + clientSecret)
+    const params = { _id: clientId }
+    if (clientSecret) params.secret = clientSecret
+    return Client.findOne(params)
+  },
+
+  getRefreshToken: refreshToken =>
+    Token.find({ refreshToken })
+      .populate('user'),
+      // .exec(),
```

Comment:
Ok so I just manually checked this and it works it the original case without `exec()` so not really sure what the problem was before. I *think* it's to do with chaining queries but it doesn't matter if you have populate at the end.